lib/git-shell-out-strategy.js

```diff
@@ -445,9 +450,9 @@ export default class GitShellOutStrategy {
     return this.gpgExec(['merge', branchName], {writeOperation: true});
   }
 
-  async isMerging() {
+  async isMerging(dotGitDir) {
```

Comment:
Interesting, I felt the opposite actually. Wanted to ensure it got kept in a single place, and the state-machine nature of the `Repository` made it easy to ensure that the value was set before we start calling other methods.